tests/cstor/volume/provision_test.go

```diff
@@ -36,53 +39,54 @@ var _ = Describe("[cstor] TEST VOLUME PROVISIONING", func() {
 
 	BeforeEach(func() {
 		When(" creating a cstor based volume", func() {
-			By("building object of storageclass")
-			scObj, err = sc.NewBuilder().
-				WithName(scName).
-				WithAnnotations(annotations).
-				WithProvisioner(openebsProvisioner).Build()
-			Expect(err).ShouldNot(HaveOccurred(), "while building storageclass obj for storageclass {%s}", scName)
-
-			By("creating storageclass")
-			_, err = ops.SCClient.Create(scObj)
-			Expect(err).To(BeNil(), "while creating storageclass", scName)
-
 			By("building spc object")
 			spcObj = spc.NewBuilder().
-				WithName(spcName).
+				WithGenerateName(spcName).
 				WithDiskType(string(apis.TypeSparseCPV)).
 				WithMaxPool(cstor.PoolCount).
 				WithOverProvisioning(false).
 				WithPoolType(string(apis.PoolTypeStripedCPV)).
 				Build().Object
 
 			By("creating storagepoolclaim")
-			_, err = ops.SPCClient.Create(spcObj)
+			spcObj, err = ops.SPCClient.Create(spcObj)
 			Expect(err).To(BeNil(), "while creating spc", spcName)
 
 			By("verifying healthy csp count")
-			cspCount := ops.GetHealthyCSPCountEventually(spcName, cstor.PoolCount)
+			cspCount := ops.GetHealthyCSPCountEventually(spcObj.Name, cstor.PoolCount)
 			Expect(cspCount).To(Equal(true), "while checking cstorpool health status")
-			//		Expect(cspCount).To(Equal(1), "while checking cstorpool health count")
+
+			By("building a CAS Config with generated SPC name")
+			CASConfig := strings.Replace(openebsCASConfigValue, "$spcName", spcObj.Name, 1)
+			CASConfig = strings.Replace(CASConfig, "$count", strconv.Itoa(cstor.ReplicaCount), 1)
+			annotations[string(apis.CASTypeKey)] = string(apis.CstorVolume)
+			annotations[string(apis.CASConfigKey)] = CASConfig
+
+			By("building object of storageclass")
+			scObj, err = sc.NewBuilder().
+				WithGenerateName(scName).
+				WithAnnotations(annotations).
+				WithProvisioner(openebsProvisioner).Build()
+			Expect(err).ShouldNot(HaveOccurred(), "while building storageclass obj for storageclass {%s}", scName)
+
+			By("creating storageclass")
+			scObj, err = ops.SCClient.Create(scObj)
+			Expect(err).To(BeNil(), "while creating storageclass", scName)
 
 		})
 	})
 
 	AfterEach(func() {
 		By("deleting resources created for testing cstor volume provisioning", func() {
+
+			By("deleting storagepoolclaim")
+			_, err = ops.SPCClient.Delete(spcObj.Name, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting the spc's {%s}", spcName)
+
 			By("deleting storageclass")
-			err = ops.SCClient.Delete(scName, &metav1.DeleteOptions{})
+			err = ops.SCClient.Delete(scObj.Name, &metav1.DeleteOptions{})
 			Expect(err).To(BeNil(), "while deleting storageclass", scName)
```

Comment:
Should it be: `Expect(err).To(BeNil(), "while deleting storageclass", scObj.Name)`